### PR TITLE
Accessory Weight Handling

### PR DESCRIPTION
**Pull request by Geevies**
* Added a system that lets accessories increase the weight class / size of the items they attach to. Currently only implemented on modular armour's arm and leg guards, which will now not fit into bags if those are attached.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Review by Arrow768 — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

File: code/__DEFINES/accessories.dm [hunk 1]
```diff
@@ -9,3 +11,8 @@
 #define ACCESSORY_SLOT_ARMOR_PIN		"armor pin"
 #define ACCESSORY_SLOT_ARMOR_POCKETS 	"armor pockets"
 #define ACCESSORY_SLOT_HEAD             "head"
+
+// Accessory Weight Classes
+#define ACCESSORY_WEIGHT_NONE 0 // doesn't increase size
+#define ACCESSORY_WEIGHT_HALF_UNIT 0.5 // two can be attached but will only increase size by one
+#define ACCESSORY_WEIGHT_ONE_UNIT 1 // will increase the size by one
```
Comment (FluffyGhoster) [inline code/__DEFINES/accessories.dm:18]: Not comments, DMDocs

File: code/modules/clothing/under/accessories/accessory.dm [hunk 2]
```diff
@@ -7,7 +7,12 @@
 	overlay_state = null
 	slot_flags = SLOT_TIE
 	w_class = ITEMSIZE_SMALL
+
 	var/slot = ACCESSORY_SLOT_GENERIC
+
+	/// When attached to another piece of clothing, it'll increase the size by this amount. the final result is rounded up
```
Comment (FluffyGhoster) [inline code/modules/clothing/under/accessories/accessory.dm:13]: Specify the defines it accepts (ACCESSORY_WEIGHT_*), and where to find them (code/__DEFINES/accessories.dm)

**Comment by Geevies** [comment]
!review

**Review by FluffyGhoster — APPROVED** [review]

**Pull request merged**